lib/cf/CloudControllerClient.js

```diff
@@ -275,6 +278,27 @@ class CloudControllerClient extends HttpClient {
     return this.getResource(`/v2/organizations/${org_guid}`, options);
   }
 
+  getMultiTenantServiceInstanceIdWithName(instance_name, org_guid, space_guid, service_name, options) {
+    return this.getResource(`/v2/spaces/${space_guid}/service_instances?q=name:${instance_name}`, options)
+      .then((serviceInstance) => {
+        if (serviceInstance.resources.length < 1) {
+          logger.debug(`Service instance with name ${instance_name} not found in the space ${space_guid}`);
+          throw new MultiTenantServiceInstanceNameNotFound(instance_name, service_name);
```

Comment:
this error's message reads `Could not find ${service} Service Instance with Name ${name} that supports multi tenancy` - however in this condition, instance with this name itself is not found. So wrong error is being thrown. Why not just throw `errors.NotFound` here?